go/engine/common.go

```diff
@@ -52,10 +52,24 @@ func findPaperKeys(ctx *Context, g *libkb.GlobalContext, me *libkb.User) (*keypa
 	if err != nil {
 		return nil, err
 	}
-	paperPhrase := libkb.NewPaperKeyPhrase(passphrase)
 
-	// the checker in GetPaperKeyPassphrase should check both of these, but
-	// just to make sure:
+	return matchPaperKey(ctx, g, me, passphrase)
+}
+
+// matchPaperKey checks to make sure paper is a valid paper phrase and that it exists
+// in the user's keyfamily.
+func matchPaperKey(ctx *Context, g *libkb.GlobalContext, me *libkb.User, paper string) (*keypair, error) {
+	cki := me.GetComputedKeyInfos()
+	if cki == nil {
+		return nil, fmt.Errorf("no computed key infos")
+	}
+	bdevs := cki.PaperDevices()
+	if len(bdevs) == 0 {
+		return nil, libkb.NoPaperKeysError{}
+	}
+
+	paperPhrase := libkb.NewPaperKeyPhrase(paper)
+
```

Comment:
I'm just going to change this to use libkb/passphrase_helper.go PaperChecker (which does that).

My concern with a check on the number of words is that if in the future we change the paper key entropy, the number of words check could fail for existing paper keys.  I guess we can cross that bridge when we come to it, but...